release-23.1: clusterversion: set developmentBranch to false

celiala opened the pull request:
This PR sets developmentBranch to false, which must be set before selected a beta.1 candidate.

Release note: none.
Epic: REL-283
Release justification: Infra-only change. Denotes `release-23.1` as a release branch/versions.

Backport 1/4 commits from #99128.

/cc @cockroachdb/release

---

As part of the 23.1 stability period tasks, this PR preps the 23.1 release branch and initializes 23.2 development, as per this release runbook: [Prep Release and Define Start of Development for Next Release](https://cockroachlabs.atlassian.net/wiki/spaces/RE/pages/2924740768/Prep+Release+and+Define+Start+of+Development+for+Next+Release).

This PR will contain 4 commits:

|commit|backported when|
|-|-|
|1. Set developmentBranch to false|soon after branch cut / before selecting beta.1 candidate|
|2. Update version.txt to alpha.8 (next release)|soon after branch cut / before selecting ~beta.1~ this week's alpha.8 candidate|
|3. Mint the previous release|should be final backport before the final RC / before selecting final RC candidate|
|4. Define the start of 23.2 development and placeholder key|_(not backported)_|


Release note: none.
Epic: REL-283

celiala commented:
> Set developmentBranch to false
This commit is causing these test to fail:
* pkg/upgrade/upgrademanager/upgrademanager_test:TestAlreadyRunningJobsAreHandledProperly
* github.com/cockroachdb/cockroach/pkg/ccl/serverccl: TestServerStartupGuardrails
And the reason isn't exactly clear to me.

Because we really need version.txt to be alpha.8 in order to select this week's candidate, I'm going to pull this into its own PR, so we can merge that sooner.

rail reviewed (changes requested):
> Reviewed 9 of 9 files at r1, 1 of 1 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/99387)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @celiala, @dt, and @renatolabs)

___
*[`docs/generated/settings/settings-for-tenants.txt` line 292 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/99387#-NRcRF2iDg8uwXGERmMv:-NRcRF2iDg8uwXGERmMw:bgj4cwk) ([raw file](https://github.com/cockroachdb/cockroach/blob/e42af560a4486a8344a55603fc726daad1f3579e/docs/generated/settings/settings-for-tenants.txt#L292)):*
> ```help
> trace.span_registry.enabled	boolean	true	if set, ongoing traces can be seen at https://<ui>/#/debug/tracez
> trace.zipkin.collector	string		the address of a Zipkin instance to receive traces, as <host>:<port>. If no port is specified, 9411 will be used.
> version	version	22.2-94	set the active cluster version in the format '<major>.<minor>'
> ```

Do you know why we set the version to `xx.yy-zz`, but mention it as `<major>.<minor>`?
___
*[`pkg/clusterversion/cockroach_versions.go` line 901 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/99387#-NRcQAq71QAamMkJw_DO:-NRcQAq71QAamMkJw_DP:b-wwwy08) ([raw file](https://github.com/cockroachdb/cockroach/blob/a7945ba77309f1eee8ce7a247d67060146e65ac4/pkg/clusterversion/cockroach_versions.go#L901)):*
> ```Go
> // 2. forced to `false` on release branches: this allows running a
> // release binary in a dev cluster.
> var developmentBranch = false || envutil.EnvOrDefaultBool("COCKROACH_FORCE_DEV_VERSION", false)
> ```

Is there any specific reason to have `false || ...` here? Looks like the first part is ignored in any case and can be removed.


<!-- Sent from Reviewable.io -->

celiala committed:
clusterversion: set developmentBranch to false

This sets developmentBranch to false for the release branch.
We plan to backport this commit as soon as the branch is cut.

Release note: none.
Epic: REL-283

celiala reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/99387)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @dt, @rail, and @renatolabs)

___
*[`docs/generated/settings/settings-for-tenants.txt` line 292 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/99387#-NRcRF2iDg8uwXGERmMv:-NSCTHsI3w0SNJEl4moC:b8rs35j) ([raw file](https://github.com/cockroachdb/cockroach/blob/e42af560a4486a8344a55603fc726daad1f3579e/docs/generated/settings/settings-for-tenants.txt#L292)):*
<details><summary><i>Previously, rail (Rail Aliiev) wrote…</i></summary><blockquote>

Do you know why we set the version to `xx.yy-zz`, but mention it as `<major>.<minor>`?
</blockquote></details>

Not sure tbh? (all I know is that this has been the format for a while :shrug: )
___
*[`pkg/clusterversion/cockroach_versions.go` line 901 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/99387#-NRcQAq71QAamMkJw_DO:-NRidgcD8S4b2VtBciD7:b9fivt1) ([raw file](https://github.com/cockroachdb/cockroach/blob/a7945ba77309f1eee8ce7a247d67060146e65ac4/pkg/clusterversion/cockroach_versions.go#L901)):*
<details><summary><i>Previously, rail (Rail Aliiev) wrote…</i></summary><blockquote>

Is there any specific reason to have `false || ...` here? Looks like the first part is ignored in any case and can be removed.
</blockquote></details>

Ack, removed `false || `


<!-- Sent from Reviewable.io -->

celiala commented:
See https://github.com/cockroachdb/cockroach/issues/100609 for failing tests

knz commented:
following up here #100685

celiala commented:
3 tests are failing here:

(1) `TestAlreadyRunningJobsAreHandledProperly`. in `pkg/upgrade/upgrademanager/upgrademanager_test`
```
=== RUN   TestAlreadyRunningJobsAreHandledProperly
    manager_external_test.go:120:
          Error Trace:  pkg/upgrade/upgrademanager/upgrademanager_test_test/pkg/upgrade/upgrademanager/manager_external_test.go:120
          Error:        Received unexpected error:
                        sql: no rows in result set
```
(2) `TestPauseMigration`. in `pkg/upgrade/upgrademanager/upgrademanager_test`
```
=== RUN   TestPauseMigration
    manager_external_test.go:464: error scanning '&{<nil> 0xc003f80200}': sql: no rows in result set
```
(3) `TestServerStartupGuardrails`. in `github.com/cockroachdb/cockroach/pkg/ccl/serverccl`
```
I230405 05:30:21.943950 189123 upgrade/upgrademanager/manager.go:168  [T1,n1] 1170  running permanent upgrades up to version: 1000022.2
I230405 05:30:21.944731 189876 sql/temporary_schema.go:505  [T1,n1] 1171  running temporary object cleanup background job
I230405 05:30:21.945053 189877 sql/sqlstats/persistedsqlstats/provider.go:129  [T1,n1] 1172  starting sql-stats-worker with initial delay: 10m15.992188656s
I230405 05:30:21.946647 189895 kv/kvclient/rangefeed/rangefeedcache/watcher.go:335  [T1,n1] 1173  system-config-cache: established range feed cache
I230405 05:30:21.947169 189864 kv/kvserver/replica_rangefeed.go:701  [T1,n1,rangefeed=table-stats-cache,mux_n=1,gen=2,r21/1:/Table/2{0-1},s1] 1174  RangeFeed closed timestamp is empty
I230405 05:30:21.950611 189876 sql/temporary_schema.go:555  [T1,n1] 1175  found 0 temporary schemas
I230405 05:30:21.950720 189876 sql/temporary_schema.go:558  [T1,n1] 1176  early exiting temporary schema cleaner as no temporary schemas were found
I230405 05:30:21.950747 189876 sql/temporary_schema.go:559  [T1,n1] 1177  completed temporary object cleanup job
I230405 05:30:21.950765 189876 sql/temporary_schema.go:640  [T1,n1] 1178  temporary object cleaner next scheduled to run at 2023-04-05 06:00:21.944702855 +0000 UTC m=+1944.305053323
E230405 05:30:21.994611 189894 spanconfig/spanconfigmanager/manager.go:141  [T1,n1] 1179  error starting auto span config reconciliation job: get-jobs: system-jobs-scan: relation "system.job_info" does not exist
I230405 05:30:22.008682 189123 upgrade/upgrademanager/manager.go:229  [T1,n1] 1180  the last permanent upgrade (v22.2-98) does not appear to have completed; attempting to run all upgrades
I230405 05:30:22.009905 189123 upgrade/upgrademanager/manager.go:270  [T1,n1] 1181  running permanent upgrade for version 0.0-2
    test_server_shim.go:331: migration-manager-find-jobs: system-jobs-scan: relation "system.job_info" does not exist
        (1) attached stack trace
          -- stack trace:
          | github.com/cockroachdb/cockroach/pkg/sql.(*InternalExecutor).execInternal.func1.1

```

celiala commented:
Fixes for failing tests have been backported 🎉 (https://github.com/cockroachdb/cockroach/issues/100685). Re-triggering CI.
